health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/presentation/exercise/ExerciseScreen.kt

```diff
@@ -81,6 +94,24 @@ fun ExerciseRoute(
     }
 }
 
+/**
+ * Shows an error that occured when starting an exercise
+ */
+@Composable
+fun ErrorStartingExerciseScreen(
+    onRestart: () -> Unit,
+    onFinishActivity: () -> Unit,
+    uiState: ExerciseScreenState
+) {
+    AlertDialog(
+        title = stringResource(id = R.string.error_starting_exercise),
+        message = "${uiState.error ?: stringResource(id = R.string.unknown_error)}. ${stringResource(id = R.string.try_again)}",
+        onCancel = onFinishActivity,
+        onOk = onRestart,
+        showDialog = uiState.error != null,
```

Comment:
nit: this is potentially ambiguous, it should be "true" as this function will not be called if uiState.error != null. If it is, then it will display a blank screen to the user.